kiva/blend2d.py

```diff
@@ -0,0 +1,780 @@
+    # Getting infomration on paths
+    # ----------------------------------------------------------------
+
+    def is_path_empty(self):
+        """ Test to see if the current drawing path is empty
+        """
+        return self.path.empty()
+
+    def get_path_current_point(self):
+        """ Return the current point from the graphics context.
+        """
+        return self.path.get_last_vertex()
+
+    def get_path_bounding_box(self):
+        """ Return the bounding box for the current path object.
+        """
+        # XXX: Returns a blend2d.Rect which is sort of useless...
+        self.path.get_bounding_box()
+
+
+    # ----------------------------------------------------------------
+    # Clipping path manipulation
+    # ----------------------------------------------------------------
+
+    def clip(self):
+        """ Clip context to a filled version of the current path.
+        """
+        raise NotImplementedError()
+
+    def even_odd_clip(self):
+        """ Clip context to a even-odd filled version of the current path.
+        """
+        raise NotImplementedError()
+
+    def clip_to_rect(self, x, y, w, h):
+        """ Clip context to the given rectangular region.
+
+            Region should be a 4-tuple or a sequence.
+        """
+        self.gc.clip_to_rect(blend2d.Rect(x, y, w, h))
+
+    def clip_to_rects(self, rects):
+        """ Clip context to a collection of rectangles
+        """
+        raise NotImplementedError()
+
+    # ----------------------------------------------------------------
+    # Color space manipulation
+    #
+    # I'm not sure we'll mess with these at all.  They seem to
+    # be for setting the color system.  Hard coding to RGB or
+    # RGBA for now sounds like a reasonable solution.
+    # ----------------------------------------------------------------
+
+    def set_fill_color_space(self):
+        msg = "set_fill_color_space not implemented for blend2d yet."
+        raise NotImplementedError(msg)
+
+    def set_stroke_color_space(self):
+        msg = "set_stroke_color_space not implemented for blend2d yet."
+        raise NotImplementedError(msg)
+
+    def set_rendering_intent(self):
+        msg = "set_rendering_intent not implemented for blend2d yet."
+        raise NotImplementedError(msg)
+
+    # ----------------------------------------------------------------
+    # Color manipulation
+    # ----------------------------------------------------------------
+
+    def set_fill_color(self, color):
+        self.gc.set_fill_style(color)
+
+    def set_stroke_color(self, color):
+        self.gc.set_stroke_style(color)
+
+    def set_alpha(self, alpha):
+        self.gc.set_alpha(alpha)
+
+    # ----------------------------------------------------------------
+    # Gradients
+    # ----------------------------------------------------------------
+
+    def linear_gradient(self, x1, y1, x2, y2, stops, spread_method,
+                        units="userSpaceOnUse"):
```

Comment:
It looks like the interface centers around the needs of agg/celiagg and some/most backends don't care about some of the arguments passed to methods on the interface - `blend2d` doesn't care about `units`.

Reply:
`units` is relevant. It's only ignored by backends because it's tricky to implement. For `"objectBoundingBox"` you need to compute the bounding box of the path being filled by the gradient and then set the transformation matrix for the fill to fit that box.